app/models/mixins/miq_request_mixin.rb

```diff
@@ -169,4 +169,10 @@ def request_dialog(action_name)
     dialog = ResourceActionWorkflow.new(values, get_user, ra, {}).dialog
     DialogSerializer.new.serialize(Array[dialog]).first
   end
+
+  def mark_execution_servers
+    options[:executed_on_servers] ||= []
+    options[:executed_on_servers] = (options[:executed_on_servers] << MiqServer.my_server.id).uniq
```

Comment:
My only issue here is the way the `.uniq` works, and the expectations on the contents of this Array.

For example, if the task executes on server `1`, then server `2`, then back to server `1`, I'm probably mostly concerned about the most recent server, so I'd expect `1` to be the last element of the Array, but...

```ruby
[1, 2, 1].uniq
# => [1, 2]
```

So, what are the expected usages of this Array?

Reply:
The initial usage is to know what servers to collect logs from.

But maybe we do not worry about calling `uniq` on the array and let it add duplicate IDs.  Then the caller can decide how they want to use the data.  A `find` call on the array will only return one instance for each ID and if you only cared about the last server you would be able to do that as well.

With retries waiting for processes to end this could become a long array, but I do not see this being a big issue.

Reply:
@Fryguy This array is to record the id of servers where the automate task has been processed. So all the logs from these servers would be collected in case of troubleshooting. The order of the ids in the array is not important in this case.

Reply:
@gmcculloug If a long array is not a concern, we can get rid of calling ```uniq```.